spec/models/reservation_spec.rb

```diff
@@ -26,4 +26,21 @@
   it 'validates raise error for invalid status' do
     expect { create(:reservation, status: 'invalid') }.to raise_error(ArgumentError, "'invalid' is not a valid status")
   end
+
+  describe 'associations' do
+    it 'validates after_create assign_available_room callback' do
```

Comment:
This is why it sounds to me more like this should be a `validation` scoped to the `Reservation` creation.